### ISSUE TITLE
Docs are missing instructions for implementing fix for lints

### ISSUE THREAD

**Issue by aminpaks** [palantir/tslint#2956]
### Help Request

- TSLint version: 5.4.3:
- Typescript version: 2.4.0:
- Using: VSCode

#### Test.ts.lint

```ts
// simple call
console.clear( );
              ~ [0]

[0]: No space within parentheses
```

#### Test.ts.fix
```ts
// simple call
console.clear();
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "space-within-parens": [true, 0]
    }
}
```

#### Expected behavior
I cannot figure out how to write fixes for my lints. I already implemented the rule and it works but when I run fix that is corresponding with the output, I always get below message and I couldn't find any docs to understand what's the problem!

```
Expected (from .fix file)
Actual (from TSLint)
// simple call
console.clear();
```

#### Fix implementation
```ts
let lintMsg: string;
let lintFix: Lint.Replacement;
if (this.options.size === 0) {
  lintMsg = Rule.FAILURE_NO_SPACE;
  lintFix = Lint.Replacement.deleteText(position, length);
}
this.addFailureAt(position, length, lintMsg, lintFix);
```

**Comment by ajafff** [comment]
Add a line break to your .fix file and it will work
The second line in your .lint file results in an empty line in the .fix output

**Comment by aminpaks** [comment]
Wonderful!

**Comment by aminpaks** [comment]
How can I PR this new rule?
What I already did:
1. Fork `tslint`
2. Added new rule
3. Compile and tests are working

**Comment by ajafff** [comment]
>We're missing this in results, an indicator may worth thousands...

I thought about that, too. I don't know if only empty lines should be handles specially or all lines...

>How can I PR this new rule?

Add it to `src/config/latest.ts` and `src/config/all.ts`
Log into https://circleci.com/ and enable test for your fork. Also set the concurrency to 4, otherwise CI will fail.

(In case you haven't done this before) Commit and push your rule to a new branch and visit https://github.com/palantir/tslint. Github will automatically suggest a PR from that branch.

**Comment by aminpaks** [comment]
Where is this `concurrency` settings?
It already fail without my commit as you said!

**Comment by ajafff** [comment]
https://circleci.com/gh/aminpaks/tslint/edit#parallel-builds

**Comment by aminpaks** [comment]
Tests on my local pass, but they failed with CI!
Do you have access to see the logs in [my branch](https://github.com/aminpaks/tslint/tree/space-within-parens) or not?

**Comment by ajafff** [comment]
CI tests with different versions of typescript. That may cause failures
even if tests pass locally.

I'll look into that tomorrow.

You can still open a PR though, so I can also give you feedback on the
implementation.

Am 26.06.2017 22:18 schrieb "Amin Pakseresht" <notifications@github.com>:

**Comment by aminpaks** [comment]
Great, thanks.
